### PR TITLE
CI Update

### PR DESCRIPTION
**Pull request by marip8**
This PR updates the CI configuration:
- Runs the build jobs on specific containers, not the Github-provided runners which are not "clean" OS images
  - Edit: apparently this is only supported for Linux jobs
- Merges the package build jobs with the regular CI jobs to eliminate code duplication

### CODE REVIEW

File: .github/workflows/ubuntu_focal.yml [hunk 1]
```diff
@@ -20,20 +26,20 @@ jobs:
...
+          rosdep init -q
           rosdep update -q
-          vcs import --input "${{ github.workspace }}/workspace/src/boost_plugin_loader/dependencies.repos" src/
+          vcs import src < src/boost_plugin_loader/dependencies.repos
           rosdep install --from-paths src --ignore-src -r -y -q
 
       - name: Build
         working-directory: workspace
         run: |
-          colcon build --event-handlers console_direct+ --cmake-args -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -DENABLE_CLANG_TIDY=ON -DENABLE_CODE_COVERAGE=ON
```
Comment (Levi-Armstrong) [inline .github/workflows/ubuntu_focal.yml:36]: When using clang-tidy and code coverage, they require debug symbols and I have had challenges with release with debug producing consistent results. This was the main reason why they have been separate CI builds.

Comment (marip8) [reply .github/workflows/ubuntu_focal.yml:36]: Interesting. I can look into setting the CMake arguments conditionally based on the event type.

Comment (marip8) [reply .github/workflows/ubuntu_focal.yml:36]: Fixed this by adding a step that will conditionally change the build type to release if it is a release job

**Comment by marip8** [comment]
@Levi-Armstrong I think this is finally ready for review

**Comment by Levi-Armstrong** [comment]
It looks like it only does a release build when a github release is triggered and all other CI builds leverage debug so clang-tidy and asserts are not affected during PR and Merges. Is that correct?

**Comment by marip8** [comment]
> It looks like it only does a release build when a github release is triggered and all other CI builds leverage debug so clang-tidy and asserts are not affected during PR and Merges. Is that correct?

Yes. It will only build in release for a release jobs; all other jobs are debug builds

**Review by Levi-Armstrong — APPROVED** [review]

**Pull request merged**